pkg/registers/registry.go

```diff
@@ -0,0 +1,40 @@
+package registers
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type registryT struct {
+	idToType map[RegisterID]reflect.Type
+}
+
+var registry = registryT{
+	idToType: make(map[RegisterID]reflect.Type),
+}
+
+// AddRegister remembers the register type and ID.
+func (registry registryT) AddRegister(reg Register) {
+	registry.idToType[reg.ID()] = reflect.TypeOf(reg)
+}
+
+// New constructs a register, given its ID and raw value. It
+// will return an error if the value is of wrong underlying type.
+func (registry registryT) New(regID RegisterID, value interface{}) (Register, error) {
```

Comment:
Why do we need it?
If one wants a register X, it can can call NewRegisterX and pass specific arguments instead of interface{}

Reply:
This is used in `UnmarshalYAML`, which has a dynamic nature.

Reply:
UnmarshalJSON didn't require to have a registry for that purpose.

I saw your comments about being able to "dynamically" parse registers.
IMHO it looks like hardcoding of adding a bit of lines to "dynamically" add register to registry vs hardcoded in switch case by id.

I'm ok to go with registry, but would prefer a bit different strategy